src/world/area_omo/omo_11/DDC830.c

```diff
@@ -51,5 +51,130 @@ ApiStatus func_80240000_DDC830(Evt* script, s32 isInitialCall) {
     return ApiStatus_BLOCK;
 }
 
+typedef struct UnkOmo11 {
+    /* 0x00 */ Vec3f unk_00;
+    /* 0x0C */ Vec3f unk_0C;
+    /* 0x18 */ Matrix4f transformMatrix;
+} UnkOmo11_2; // size = 0x58
 
-INCLUDE_ASM(s32, "world/area_omo/omo_11/DDC830", func_802401A0_DDC9D0);
+extern s32 D_80241760_DDDF90[6]; // TODO remove hard-coded size once data is migrated
+extern s32 D_80241778_DDDFA8[6]; // TODO remove hard-coded size once data is migrated
+
+NOP_FIX
+ApiStatus func_802401A0_DDC9D0(Evt* script, s32 isInitialCall) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Matrix4f sp20, sp60, spA0, spE0;
+    Npc* partner = get_npc_unsafe(NPC_PARTNER);
+    Model* model1 = get_model_from_list_index(get_model_list_index_from_tree_index(0x4A));
+    Model* model2 = get_model_from_list_index(get_model_list_index_from_tree_index(0x3F));
+    f32 ox, oy, oz;
+    Model* loopModel;
+    f32 x, y, z;
+    UnkOmo11_2* it;
+    u32 cond;
+    u32 i;
+    s32 j, k;
+
+    if (isInitialCall) {
+        sfx_play_sound_at_position(SOUND_80000016, 0, 315.0f, 125.0f, -100.0f);
+        script->functionTempPtr[0] = it = heap_malloc(sizeof(*it) * ARRAY_COUNT(D_80241760_DDDF90));
+        script->functionTemp[1] = 0;
+
+        for (i = 0; i < ARRAY_COUNT(D_80241760_DDDF90); it++, i += 2) {
+            loopModel = get_model_from_list_index(get_model_list_index_from_tree_index(D_80241760_DDDF90[i]));
+            it->unk_00.x = x = loopModel->center.x - model1->center.x;
+            it->unk_00.y = y = loopModel->center.y - model1->center.y;
+            it->unk_00.z = z = loopModel->center.z - model1->center.z;
+
+            ox = x;
+            oy = y;
+            oz = z;
+
+            for (j = 0; j < 4; j++) {
+                for (k = 0; k < 4; k++) {
+                    it->transformMatrix[j][k] = loopModel->transformMatrix[j][k];
+                }
+            }
+
+            loopModel = get_model_from_list_index(get_model_list_index_from_tree_index(D_80241760_DDDF90[i + 1]));
+
+            it = it + 1;
+            it->unk_00.x = ox;
+            it->unk_00.y = oy;
+            it->unk_00.z = oz;
+
+            for (j = 0; j < 4; j++) {
+                for (k = 0; k < 4; k++) {
+                    it->transformMatrix[j][k] = loopModel->transformMatrix[j][k];
+                }
+            }
+        }
+    }
+
+    it = script->functionTempPtr[0];
+    for (i = 0; i < 6; i++, it++) {
+        loopModel = get_model_from_list_index(get_model_list_index_from_tree_index(D_80241760_DDDF90[i]));
+
+        for (j = 0; j < 4; j++) {
+            for (k = 0; k < 4; k++) {
```

Comment:
Worth ARRAY_COUNTing these to be the transformMatrix lengths probably

Reply:
I think affine transform matrices can always be assumed to be 4x4; this can go either way imo.

Reply:
yeah idk, I don't think array_count is warranted here. if anyone is changing the dimensions of Matrix4f, good luck to them